ndcube/ndcube.py

```diff
@@ -964,24 +966,98 @@ def _new_instance(self, **kwargs):
+                kwargs["uncertainty"] = self.combine_uncertainty(value, result_data)
+
+            elif (self_unmasked and not value_unmasked):
+                kwargs["mask"] = value.mask # mask needs to be set.
+                # combine the uncertainty
+                kwargs["uncertainty"] = self.combine_uncertainty(value, result_data)
+
+            elif (value_unmasked and not self_unmasked):
+                kwargs["mask"] = self.mask
+                # combine the uncertainty
+                kwargs["uncertainty"] = self.combine_uncertainty(value, result_data)
+
+            else:
+                kwargs["mask"] = handle_mask(self.mask,value.mask)
+                # combine the uncertainty
+                kwargs["uncertainty"] = self.combine_uncertainty(value, result_data)
+
+
+        elif hasattr(value, 'unit'):
             if isinstance(value, u.Quantity):
                 # NOTE: if the cube does not have units, we cannot
                 # perform arithmetic between a unitful quantity.
                 # This forces a conversion to a dimensionless quantity
                 # so that an error is thrown if value is not dimensionless
                 cube_unit = u.Unit('') if self.unit is None else self.unit
-                new_data = self.data + value.to_value(cube_unit)
+                kwargs["data"] = self.data + value.to_value(cube_unit)
             else:
                 # NOTE: This explicitly excludes other NDCube objects and NDData objects
                 # which could carry a different WCS than the NDCube
                 return NotImplemented
         elif self.unit not in (None, u.Unit("")):
             raise TypeError("Cannot add a unitless object to an NDCube with a unit.")
         else:
-            new_data = self.data + value
-        return self._new_instance(data=new_data)
+            kwargs["data"] = self.data + value
+
+        # return the new NDCube instance
+        return self._new_instance(**kwargs)
+
+    def __add__(self, value):
+        # when value has a mask, raise error and point user to the add method. TODO
+        #
+        # check whether there is a mask.
+        # Neither self nor value has a mask
+
+        self_masked = not(self.mask is None or self.mask is False or not self.mask.any()) if hasattr(self, "mask") else False
+        value_masked = not(value.mask is None or value.mask is False or not value.mask.any()) if hasattr(value, "mask") else False
+
+        if  (value_masked or (self_masked and hasattr(value,'uncertainty') and value.uncertainty is not None)): # value has a mask,
+            # let the users call the add method, since the handle_mask keyword cannot be given by users here.
+            raise TypeError('Please use the add method.')
+
+        return self.add(value) # without any mask, the add method can be called here and will work properly without needing arguments to be passed.
+
+    def combine_uncertainty(self, value, result_data):
```

Comment:
This should be a hidden method, i.e. not intended for users.  Therefore, it should start with an `_`
```suggestion
    def _combine_uncertainty(self, value, result_data):
```